File: Dailo/Dailo.ClientApp/src/app/modules/habit-entries/models/habit-entry.model.ts
import { HabitEntrySource } from '@habit-entries/enums/habit-entry-source.enum';
import { HabitType } from '@habits/enums/habit-type.enum';

export interface HabitEntryModel {
  id: number;
  habitId: string;
  habitName: string;
  habitType: HabitType;
  value: number;
  notes: string | null;
  source: HabitEntrySource;
  externalId: string | null;
  completedAtUtc: string;
  isArchived: boolean;
  createdAtUtc: Date;
  updatedAtUtc: Date;
}

File: Dailo/HabitEntry.Application/Models/HabitEntryModel.cs
using HabitEntry.Domain.Enums;
using StrictId;

namespace HabitEntry.Application.Models;

public class HabitEntryModel
{
    public required Id<HabitEntryModel> Id { get; set; }

    public required Id HabitId { get; set; }

    public required string HabitName { get; set; }

    public required int Value { get; set; }

    public string? Notes { get; set; }

    public required HabitEntrySource Source { get; set; }

    public string? ExternalId { get; set; }

    public required DateTime CompletedAtUtc { get; set; }

    public bool IsArchived { get; set; }

    public DateTime CreatedAtUtc { get; set; }

    public DateTime? LastModifiedAtUtc { get; set; }
}
